lib/puppet/resource/type.rb

```diff
@@ -360,9 +360,6 @@ def set_resource_parameters(resource, scope)
     resource.add_parameters_from_consume
     inject_external_parameters(resource, scope)
 
-    resource_hash = {}
```

Comment:
Since this is changes in resource/types.rb (that also is used client side), it requires extra caution. Now, this happens to be in a method that only makes sense on the compiling side. I think we should update the method documentation to state that (makes the next person working on this code less anxious about making changes). (Github did not allow me to add a note on the actual comment for this method, so I inserted the comment here.